### ISSUE TITLE
Mismatched lists length when using predict function

### ISSUE THREAD

**Issue by wshuyi** [ThilinaRajapakse/simpletransformers#11]
I tried to use the `model.predict()` function. The length of the input list (of strings) is 400, yet the length of predictions is only 320. May I know what is wrong? Besides, would you please add `accuracy` and `f1 score` as default metrics too? Thanks!

**Comment by ThilinaRajapakse** [comment]
I can't reproduce the error with the length of predictions. I get a list of 400 predictions for an input list of length 400.

I added the option to pass in custom metrics. You can pass in additional metrics as keyword arguments like this.

```
result, model_outputs, wrong_predictions = model.eval_model(eval_df, acc=sklearn.metrics.accuracy_score)
```

**Comment by samarthsarin** [comment]
So the accuracy will be stored under the model_outputs variable?

**Comment by ThilinaRajapakse** [comment]
No, it will be added to the `results` dictionary along with all other metrics.

**Comment by samarthsarin** [comment]
Ok thank you

**Comment by samarthsarin** [comment]
Other than accuracy can we add classification report and f1 score as well?

**Comment by ThilinaRajapakse** [comment]
Yeah, you can add as many as you want.

**Comment by wshuyi** [comment]
Thanks a lot for the quick response! I did eval first, and the length of valid dataset is 320. The length of test set is 400. Is there any probability that when I called predict, instead of using test set I input, the model still memorized and used valid set?

**Comment by wshuyi** [comment]
I just removed the evaluation part and did prediction directly. With exactly the same code, I got the result list as length 400. I am afraid there is a bug in the prediction part.

**Comment by ThilinaRajapakse** [comment]
Oh, I got it. I wasn't able to reproduce the issue because I was using `reprocess_input_data = True`. 

> Is there any probability that when I called predict, instead of using test set I input, the model still memorized and used valid set?

This was exactly what was going on. The predict function was picking up on the cached evaluate features. I think it should be fixed now. Can you try it and let me know, please?

**Comment by wshuyi** [comment]
Worked perfectly this time. Thanks a lot!
